libvast/vast/data.hpp

```diff
@@ -55,7 +55,11 @@ using to_data_type = std::conditional_t<
     bool,
     std::conditional_t<
       std::is_unsigned_v<T>,
-      count,
+      std::conditional_t<
+        sizeof(T) == 1,
```

Comment:
Isn't this a bit brittle? What is we construct data from a `char`? Shouldn't this convert to a string? It would be an enum now.

(This is exacerbated by the non-explicit constructor in `data`.)

Reply:
I forgot to add a comment here. I can't think of a way to do this properly other then converting the primitive types to structs.

Reply:
Yeah, adding a comment here strikes me as the right way to handle it within the context of this PR.